### PR TITLE
Add additional services specific controls. Closes #59

### LINKED ISSUE #59: Add additional services specific controls
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**
1. [x] vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled
2. [x] vpc_network_firewall_deletion_protection_enabled
3. [x] vpc_network_firewall_policy_encrypted_with_kms_cmk
4. [x] vpc_network_firewall_rule_group_encrypted_with_kms_cmk
5. [x] vpc_network_firewall_encrypted_with_kms_cmk
6. [x] vpc_endpoint_service_acceptance_enabled
7. [x] vpc_transfer_server_not_publicly_accesible
8. [x] redshift_cluster_encryption_enabled
9. [x] redshift_serverless_namespace_encrypted_with_kms_cmk
10. [x] redshift_snapshot_copy_grant_encrypted_with_kms_cmk
11. [x] rds_db_cluster_instance_performance_insights_enabled
12. [x] rds_db_cluster_instance_performance_insights_encrypted_with_kms_cmk
13. [x] rds_db_instance_performance_insights_encrypted_with_kms_cmk
14. [x] rds_db_instance_performance_insights_enabled
15. [x] rds_cluster_activity_stream_encrypted_with_kms_cmk
16. [x] rds_db_instance_uses_recent_ca_cert
17. [x] memorydb_snapshot_encrypted_with_kms_cmk
18. [x] memorydb_cluster_encrypted_with_kms_cmk
19. [x] memorydb_cluster_transit_encryption_enabled
20. [x] rds_db_snapshot_not_publicly_accesible
21. [x] rds_db_snapshot_copy_encrypted_with_kms_cmk
22. [x] rds_db_cluster_instance_automatic_minor_version_upgrade_enabled
23. [x] rds_db_cluster_encryption_enabled
24. [x] rds_mysql_db_cluster_audit_logging_enabled
25. [x] efs_access_point_has_user_identity
26. [x] efs_access_point_has_root_directory
27. [x] codebuild_project_s3_logs_encryption_enabled
28. [x] codebuild_project_logging_enabled
29. [x] kinesis_firehose_delivery_stream_server_side_encryption_enabled
30. [x] kinesis_firehose_delivery_stream_encrypted_with_kms_cmk
31. [x] kinesis_stream_encrypted_with_kms_cmk
32. [x] kinesis_video_stream_encrypted_with_kms_cmk
33. [x] cloudtrail_trail_sns_topic_enabled
34. [x] cloudtrail_event_data_store_encrypted_with_kms_cmk

### CODE REVIEW

**Review by rajlearner17 — APPROVED** [review]
LGTM

File: query/rds.sp [hunk 1]
```diff
@@ -136,6 +136,94 @@ query "rds_db_cluster_multiple_az_enabled" {
...
+  sql = <<-EOQ
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments -> 'performance_insights_kms_key_id') is null then 'alarm'
+        else 'ok'
+      end status,
+      name || case
+        when (arguments -> 'performance_insights_kms_key_id') is null then ' ''performance_insights_kms_key_id'' not set'
+        else ' ''performance_insights_kms_key_id'' set'
```
Comment (madhushreeray30) [inline query/rds.sp:172]: We should mention > `performance insights encrypted with KMS CMK` and `performance insights not encrypted with KMS CMK`

Comment (madhushreeray30) [reply query/rds.sp:172]: please make similar changes everywhere else

File: query/rds.sp [hunk 2]
```diff
@@ -136,6 +136,94 @@ query "rds_db_cluster_multiple_az_enabled" {
...
+      type || ' ' || name as resource,
+      case
+        when (arguments -> 'performance_insights_enabled') is null then 'alarm'
+        when (arguments -> 'performance_insights_enabled')::boolean then 'ok'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments -> 'performance_insights_enabled') is null then ' ''performance_insights_enabled'' disabled'
+        when (arguments -> 'performance_insights_enabled')::boolean then ' ''performance_insights_enabled'' enabled'
+        else ' ''performance_insights_enabled'' disabled'
```
[the remark this answers is not in the document]
Comment (madhushreeray30) [reply query/rds.sp:151]: please make similar changes everywhere else

File: query/rds.sp [hunk 3]
```diff
@@ -581,3 +690,196 @@ query "rds_db_security_group_events_subscription" {
...
+query "memorydb_cluster_transit_encryption_enabled" {
+  sql = <<-EOQ
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments ->> 'tls_enabled') = 'false' then 'alarm'
+        else 'ok'
+      end status,
+      name || case
+        when (arguments ->> 'tls_enabled') = 'false' then ' transit encryption not enabled'
```
Comment (madhushreeray30) [inline query/rds.sp:768]:
```suggestion
        when (arguments ->> 'tls_enabled') = 'false' then ' encryption at transit disabled'
```

File: query/rds.sp [hunk 4]
```diff
@@ -581,3 +690,196 @@ query "rds_db_security_group_events_subscription" {
...
+        else ' transit encryption enabled'
```
Comment (madhushreeray30) [inline query/rds.sp:769]:
```suggestion
        else ' encryption at transit enabled'
```

File: query/rds.sp [hunk 5]
```diff
@@ -581,3 +690,196 @@ query "rds_db_security_group_events_subscription" {
...
+query "rds_db_cluster_instance_automatic_minor_version_upgrade_enabled" {
+  sql = <<-EOQ
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments ->> 'auto_minor_version_upgrade') = 'false' then 'alarm'
+        else 'ok'
+      end status,
+      name || case
+        when (arguments ->> 'auto_minor_version_upgrade') = 'false' then ' auto minor version upgrade not enabled'
```
Comment (madhushreeray30) [inline query/rds.sp:831]:
```suggestion
        when (arguments ->> 'auto_minor_version_upgrade') = 'false' then ' auto minor version upgrade disabled'
```

File: query/vpc.sp [hunk 6]
```diff
@@ -204,4 +204,157 @@ query "vpc_subnet_auto_assign_public_ip_disabled" {
...
+query "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
+  sql = <<-EOQ
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments ->> 'auto_accept_shared_attachments') = 'enable' then 'alarm'
+        else 'ok'
+      end status,
+      name || case
+        when (arguments ->> 'auto_accept_shared_attachments') = 'enable' then ' ''auto_accept_shared_attachments'' enabled'
```
Comment (madhushreeray30) [inline query/vpc.sp:350]:
```suggestion
        when (arguments ->> 'auto_accept_shared_attachments') = 'enable' then ' automatically accept VPC attachment requests.'
```

File: query/vpc.sp [hunk 7]
```diff
@@ -204,4 +204,157 @@ query "vpc_subnet_auto_assign_public_ip_disabled" {
...
+        else ' ''auto_accept_shared_attachments'' not enabled'
```
Comment (madhushreeray30) [inline query/vpc.sp:351]:
```suggestion
        else ' do not automatically accept VPC attachment requests'
```

**Review by madhushreeray30 — COMMENTED** [review]
@Priyanka-Chatterjee-2000 please have a look at the comments, thank!

File: controls/cloudtrail.sp [hunk 8]
```diff
@@ -63,3 +65,19 @@ control "cloudtrail_trail_validation_enabled" {
...
+control "cloudtrail_trail_sns_topic_enabled" {
+  title       = "CloudTrail trail should have SNS topic enabled"
+  description = "Enable SNS topic for CloudTrail trails to notify when new log files are delivered."
+  query       = query.cloudtrail_trail_sns_topic_enabled
+
+  tags = local.cloudtrail_compliance_common_tags
+}
+
+control "cloudtrail_event_data_store_encrypted_with_kms_cmk" {
+  title      = "CloudTrail event data should be stored encrypted with KMS CMK"
```
Comment (madhushreeray30) [inline controls/cloudtrail.sp:78]:
```suggestion
  title       = "CloudTrail event data should be stored encrypted with KMS CMK"
```

File: query/vpc.sp [hunk 9]
```diff
@@ -204,4 +204,157 @@ query "vpc_subnet_auto_assign_public_ip_disabled" {
...
+query "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
+  sql = <<-EOQ
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments ->> 'auto_accept_shared_attachments') = 'enable' then 'alarm'
+        else 'ok'
+      end status,
+      name || case
+        when (arguments ->> 'auto_accept_shared_attachments') = 'enable' then ' automatically accept VPC attachment requests.'
```
Comment (madhushreeray30) [inline query/vpc.sp:350]:
```suggestion
        when (arguments ->> 'auto_accept_shared_attachments') = 'enable' then ' automatically accept VPC attachment requests'
```

File: query/rds.sp [hunk 10]
```diff
@@ -136,6 +136,94 @@ query "rds_db_cluster_multiple_az_enabled" {
...
+query "rds_db_cluster_instance_performance_insights_encrypted_with_kms_cmk" {
+  sql = <<-EOQ
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments -> 'performance_insights_kms_key_id') is null then 'alarm'
+        else 'ok'
+      end status,
+      name || case
+        when (arguments -> 'performance_insights_kms_key_id') is null then ' ''performance_insights_kms_key_id'' not set'
```
Comment (madhushreeray30) [inline query/rds.sp:171]:
```suggestion
        when (arguments -> 'performance_insights_kms_key_id') is null then ' performance insights not encrypted with KMS CMK'
```

File: query/rds.sp [hunk 11]
```diff
@@ -136,6 +136,94 @@ query "rds_db_cluster_multiple_az_enabled" {
...
+        else ' ''performance_insights_kms_key_id'' set'
```
Comment (madhushreeray30) [inline query/rds.sp:172]:
```suggestion
        else ' performance insights encrypted with KMS CMK'
```

File: controls/cloudtrail.sp [hunk 12]
```diff
@@ -63,3 +65,19 @@ control "cloudtrail_trail_validation_enabled" {
...
+  title       = "CloudTrail trail should have SNS topic enabled"
+  description = "Enable SNS topic for CloudTrail trails to notify when new log files are delivered."
+  query       = query.cloudtrail_trail_sns_topic_enabled
+
+  tags = local.cloudtrail_compliance_common_tags
+}
+
+control "cloudtrail_event_data_store_encrypted_with_kms_cmk" {
+  title       = "CloudTrail event data should be stored encrypted with KMS CMK"
+  description = "This control checks whether CloudTrail event data is stored encrypted with KMS CMK."
```
Comment (misraved) [inline controls/cloudtrail.sp:79]:
```suggestion
  description = "This control checks whether a CloudTrail event data store is encrypted with KMS CMK."
```

**Pull request merged**